pulsar-broker/src/main/java/org/apache/pulsar/broker/systopic/SystemTopicClient.java

```diff
@@ -168,12 +172,19 @@
     }
 
     static boolean isSystemTopic(TopicName topicName) {
-        if (topicName.isPartitioned()) {
-            return EventsTopicNames.NAMESPACE_EVENTS_LOCAL_NAME
-                    .equals(TopicName.get(topicName.getPartitionedTopicName()).getLocalName());
+        if (Objects.equal(SYSTEM_NAMESPACE, topicName.getNamespaceObject())) {
+            return true;
         }
+        String localName = TopicName.get(topicName.getPartitionedTopicName()).getLocalName();
+        if (StringUtils.equals(localName, EventsTopicNames.NAMESPACE_EVENTS_LOCAL_NAME) || StringUtils
+                .equals(localName, EventsTopicNames.TRANSACTION_BUFFER_SNAPSHOT) || (
+                StringUtils.startsWith(localName, EventsTopicNames.TRANSACTION_BUFFER_SNAPSHOT) && StringUtils
+                        .endsWith(localName, MLPendingAckStore.PENDING_ACK_STORE_SUFFIX))) {
```

Comment:
This does not seem to be a good way. Whenever a new SystemTopic is added, it must be changed here.

Reply:
Good idea, add the EVENTS_TOPICS set.